web/src/bootstrap_typeahead.ts

```diff
@@ -658,6 +658,10 @@ export class Typeahead<ItemType extends string | object> {
                 ) {
                     return;
                 }
+                // Prevent typeahead from showing when deleting a pill with backspace.
+                if (pseudo_keycode === 8 ){
+                    return;
+                }
```

Comment:
@evykassirer is this a better approach or should I pass `hideOnEmpty` in `TypeaheadOptions `?

Reply:
The difference between this and `calling this.lookup(true)` (through `hideOnEmpty`) is `lookup` will close the typeahead if it's already open and only close it if the query is shorter than `MIN_LENGTH`.

If you do this instead, the typeahead might not close if it's already open when deleting a pill, and it also might not open when typing backspace in the middle of typing a regular query. It would be good for you to check out those two cases and see what happens.

And we should also decide if we always want this change for every typeahead, or only for the invite modal. And if folks decide we do want it for every typeahead, you'll want to test the other typeaheads to make sure they work okay.

Reply:
> If you do this instead, the typeahead might not close if it's already open when deleting a pill, and it also might not open when typing backspace in the middle of typing a regular query.

You were right about it.

The PR now uses `hideOnEmpty` method.

Reply:
> And we should also decide if we always want this change for every typeahead, or only for the invite modal. And if folks decide we do want it for every typeahead, you'll want to test the other typeaheads to make sure they work okay.

[CZO ](https://chat.zulip.org/#narrow/stream/101-design/topic/pill.20typeahead.20backspace.20interaction/near/1838234)to answer this.